scipy/spatial/distance.py

```diff
@@ -1468,6 +1419,7 @@ def pdist(X, metric='euclidean', p=2, w=None, V=None, VI=None):
         if w is not None and not mstr.startswith("test_"):
             if mstr in ['seuclidean', 'se', 's', 'mahalanobis']:
                 raise ValueError("metric %s incompatible with weights" % mstr)
+            # need to use python version for weighting
```

Comment:
Sry, I didn't explain myself. I meant to add a suggestion in the Exception message...something like "use directly metric XXX or call with 'test_XXX'"